subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -348,7 +395,7 @@ static void test_finalize(void)
 
 void ztest_test_fail(void)
 {
-	test_result = ZTEST_RESULT_FAIL;
+	test_result = (phase == TEST_PHASE_SETUP) ? ZTEST_RESULT_SUITE_FAIL : ZTEST_RESULT_FAIL;
 	test_finalize();
```

Comment:
This function is being invoked by the assert during setup.  `test_finalize` kills the current thread.  I think we need to conditionally call `test_finalize` when `phase != TEST_PHASE_SETUP`

Reply:
Thanks, I see the same below for skip.